utils/guardrails/guard.py

```diff
@@ -0,0 +1,204 @@
+        return guardrails, guardrails_str
+    
+    def set_llm(self, api: str, params: Optional[Dict]):
+        """
+        Set the LLM based on the provided API and parameters.
+
+        Parameters:
+        - api (str): The LLM API to use, either 'sambaverse' or 'sambastudio'.
+        - params (Dict, optional): Additional parameters for the LLM.
+
+        Returns:
+        - LLM: An instance of the specified LLM.
+
+        Raises:
+        - ValueError: If an invalid API is provided.
+        """
+        if api == "sambastudio":
+            llm = SambaStudio(
+                **params,
+                model_kwargs={
+                "select_expert": "Meta-Llama-Guard-2-8B",
+                "process_prompt": False,
+                "do_sample": False,
+                "max_tokens_to_generate": 1024,
+                "temperature": 0.1,
+                },
+            )
+        elif api == "sambaverse":
+            llm = Sambaverse(
+                **params,
+                sambaverse_model_name="Meta/Meta-Llama-Guard-2-8B",
+                model_kwargs={
+                    "select_expert": "Meta-Llama-Guard-2-8B",
+                    "process_prompt": True,
+                    "do_sample": False,
+                    "max_tokens_to_generate": 1024,
+                    "temperature": 0.1,
+                },
+            )
+        else:
+            raise ValueError(f"Invalid LLM API: {api}, only'sambastudio' and'sambaverse' are supported.")
+        return llm
+
+    def evaluate(
+        self,
+        input: Union[List[Dict], str],
+        role: str,
+        error_message: str = None,
+        return_guardrail_type: bool = True,
+        raise_exception: bool =False
+        ) -> Union[str, List[str]]: 
+        """
+        Evaluate a message or a conversation against the guardrails.
+
+        Parameters:
+        - input (Union[List[Dict], str]): The message or conversation to evaluate. It can be a string with the message or a list of dictionaries representing a conversation.
+            Example conversation input
+            [
+             {"message_id":0,"role":"user", "content":"this is an user message"},
+             {"message_id":1,"role":"assistant","content":"this is an assistant response"},
+            ]
+        - role (str): The role of the message to analyse. It can be either 'user' or 'assistant'.
+        - error_message (str, optional): The error message to be displayed when the message violates the guardrails. If not provided, a default message "The message violate guardrails" will be used.
+        - return_guardrail_type (bool, optional): If True, the function will return the violated guardrail types. If False, it will only return the error message. Default is True.
+        - raise_exception (bool, optional): If True, the function will raise a ValueError with the error message when the message violates the guardrails. If False, it will return the error message. Default is False.
```

Comment:
If False, it will return the error message without raising the exception.

Reply:
added